presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomFilters.java

```diff
@@ -0,0 +1,333 @@
+        CodedInputStream input = CodedInputStream.newInstance(bytes);
+        OrcProto.BloomFilterIndex deserializedBloomFilterIndex = OrcProto.BloomFilterIndex.parseFrom(input);
+        List<OrcProto.BloomFilter> bloomFilterList = deserializedBloomFilterIndex.getBloomFilterList();
+        assertEquals(bloomFilterList.size(), 1);
+
+        OrcProto.BloomFilter bloomFilterRead = bloomFilterList.get(0);
+
+        // Validate contents of ORC bloom filter bitset
+        compareBitset(Longs.toArray(bloomFilterRead.getBitsetList()), bloomFilterWrite.getBitSet());
+
+        // hash functions
+        assertEquals(bloomFilterWrite.getNumHashFunctions(), bloomFilterRead.getNumHashFunctions());
+
+        // bit size
+        assertEquals(bloomFilterWrite.getBitSet().length, bloomFilterRead.getBitsetCount());
+    }
+
+    private static OrcProto.BloomFilter toOrcBloomFilter(BloomFilter bloomFilter)
+    {
+        OrcProto.BloomFilter.Builder builder = OrcProto.BloomFilter.newBuilder();
+        for (long bits : bloomFilter.getBitSet()) {
+            builder.addBitset(bits);
+        }
+        builder.setNumHashFunctions(bloomFilter.getNumHashFunctions());
+        return builder.build();
+    }
+
+    @Test
+    public void testBloomFilterPredicateValuesExisting()
+            throws Exception
+    {
+        ImmutableMap<Object, Type> writeValues = getTestElements();
+
+        BloomFilter bloomFilter = new BloomFilter(writeValues.size() * 10, 0.01D);
+
+        for (Object o : writeValues.keySet()) {
+            if (o instanceof Long) {
+                bloomFilter.addLong((Long) o);
+            }
+            else if (o instanceof Integer) {
+                bloomFilter.addLong((Integer) o);
+            }
+            else if (o instanceof String) {
+                bloomFilter.addString((String) o);
+            }
+            else if (o instanceof BigDecimal) {
+                bloomFilter.addString(o.toString());
+            }
+            else if (o instanceof Slice) {
+                bloomFilter.addString(((Slice) o).toStringUtf8());
+            }
+            else if (o instanceof Timestamp) {
+                bloomFilter.addLong(((Timestamp) o).getTime());
+            }
+            else if (o instanceof Double) {
+                bloomFilter.addDouble((Double) o);
+            }
+            else {
+                fail("Unsuppored type " + o.getClass());
+            }
+        }
+
+        for (Map.Entry<Object, Type> testValue : writeValues.entrySet()) {
+            boolean matched = TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, testValue.getKey(), testValue.getValue());
+            assertTrue(matched, "type " + testValue.getClass());
+        }
+
+        // test unsupported type
+        assertTrue(TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, new Date(), TIMESTAMP), "unsupported type Date should always return true");
```

Comment:
DATE (and I believe TIMESTAMP) is supported for ORC bloom filters, but you don't need to implement this now.  Just file an issue